src/contrib/mlir/backend/analysis/memory_analysis.cpp

```diff
@@ -68,16 +68,16 @@ namespace
     {
     public:
         /// Initialize the relationship for a number of syms
-        void init(std::unordered_set<Value*>& symbols);
+        void init(DenseSet<Value>& symbols);
         /// Checks if values a and b can alias
-        bool canAlias(Value* a, Value* b);
-        void insertNoAlias(Value* a, Value* b);
+        bool canAlias(Value a, Value b);
+        void insertNoAlias(Value a, Value b);
 
     private:
         using BV = llvm::BitVector;
-        std::unordered_map<Value*, unsigned> m_valueToIdx;
-        std::unordered_map<unsigned, Value*> m_idxToValue;
-        std::unordered_map<Value*, BV*> m_valueToSet;
+        DenseMap<Value, unsigned> m_valueToIdx;
```

Comment:
`unordered_map --> DenseMap` is needed for MLIR upgarde ?

Reply:
Yes, basically:
> We should used LLVM container as much as possible since MLIR implements hashes and other utilities needed to manipulate these wrapper classes in containers.

Value is no longer a pointer so we would need to implement a hash for it. MLIR already has all the utilities needed for DenseMap.